ui/test/check_includes_ui.sh

```
@@ -0,0 +1,53 @@
+# check_includes_ui.sh — the UI layer's missing-#include preflight.
+#
+# forge-kernel/test/native/check_includes.sh covers the libc++-supplies-it /
+# libstdc++-does-not traps that have actually broken CI on this repo. The UI
+# layer leans on a wider slice of the standard library (containers, optional,
+# functional, iosfwd), so this is a SUPERSET for ui/ only — extending the kernel
+# script would change what the kernel gate accepts, which is not this file's job.
+#
+# Usage: bash check_includes_ui.sh [file ...]   (no args = every ui/ file)
+# bash-3.2 compatible (macOS default) — no associative arrays.
+set -uo pipefail
+cd "$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." && pwd)"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)